packages/adminrouter/build

```diff
@@ -27,86 +27,34 @@ make install
 popd
 
 # Incorporate adminrouter components.
-# TODO(jp): Currently, we need to support multiple build variants
-# from the same build script which seems unclean. Also, shouldn't
-# we just copy the entire AR repo content instead of cherry-picking?
 pushd "/pkg/src/adminrouter"
 
 nginx_config_path="$PKG_PATH/nginx/conf"
 mkdir -p $nginx_config_path
-cp nginx.conf $nginx_config_path/
-cp snakeoil.crt $nginx_config_path/
-cp snakeoil.key $nginx_config_path/
-cp *.lua $nginx_config_path/
-if [ -d errorpages ]; then
-    cp -r errorpages $nginx_config_path/
-fi
-# Copy OpenResty Lua modules.
-if [ -d resty ]; then
-    cp -r resty $nginx_config_path/
-fi
+cp -r * $nginx_config_path/
 
 popd
 
 # Copy  in needed libraries
 mkdir -p "$PKG_PATH/lib"
 cp /lib/x86_64-linux-gnu/libpcre.so.3 "$PKG_PATH/lib/libpcre.so.3"
 
-# From: https://projects.archlinux.org/svntogit/packages.git/tree/trunk/service?h=packages/nginx
 systemd_master="$PKG_PATH"/dcos.target.wants_master/dcos-adminrouter.service
 mkdir -p "$(dirname "$systemd_master")"
-cat <<EOF > "$systemd_master"
-[Unit]
-Description=Admin Router: A high performance web server and a reverse proxy server
-After=dcos-gen-resolvconf.service
+envsubst '$PKG_PATH' < /pkg/extra/dcos-adminrouter.service > "$systemd_master"
 
-[Service]
-Restart=always
-StartLimitInterval=0
-RestartSec=5
-EnvironmentFile=/opt/mesosphere/environment
-Type=forking
-PIDFile=$PKG_PATH/nginx/logs/nginx.pid
-PrivateDevices=yes
-StandardOutput=journal
-StandardError=journal
-ExecStartPre=/bin/ping -c1 ready.spartan
-ExecStartPre=/bin/ping -c1 marathon.mesos
-ExecStartPre=/bin/ping -c1 leader.mesos
-ExecStartPre=/usr/bin/test -f /var/lib/dcos/cluster-id
-ExecStart=$PKG_PATH/nginx/sbin/nginx
-ExecReload=/usr/bin/kill -HUP \$MAINPID
-KillSignal=SIGQUIT
-KillMode=mixed
-EOF
+systemd_slave="$PKG_PATH"/dcos.target.wants_slave/dcos-adminrouter-slave.service
```

Comment:
Not to beat a dead horse, but should we be using `agent` instead of `slave` terminology here? @spahl ?